compiler/src/iree/compiler/Codegen/Common/TransformDialectStrategies.cpp

```diff
@@ -296,7 +303,8 @@ static ReductionSplitResult createExpansionBubbleUp(
   }
 
   auto funcH = b.create<MatchOp>(variantH, func::FuncOp::getOperationName());
-  auto applyPatterns = b.create<ApplyPatternsOp>(funcH, /*rankReducing=*/false);
+  auto applyPatterns = b.create<ApplyPatternsOp>(
+      funcH, /*rankReducing=*/false, /*foldReassociativeReshapes=*/false);
```

Comment:
I was thinking it is tricky to determine what to do where if we have such closely tied cases of when to activate rank-reducing vs foldReassociative vs bubblecollapse but I realize all this complexity goes away with a proper struct in the builder.

In each of those cases, you only activate 1 type of pattern, it's just the builder API that makes it confusing what one needs to deactivate (nothing) and what one needs to activate (one thing).